src/core/common/instance.hpp

```diff
@@ -277,10 +258,7 @@ class Instance : public otInstance, private NonCopyable
      *
      */
     Utils::Heap &GetHeap(void) { return mHeap; }
```

Comment:
No other module should be using this method. If they does, the code is not working when external heap is enabled. So, I would suggest to remove `GetHeap()` method in the first place to avoid potential issues.

Reply:
This makes sense to me, but unfortunately, there is actually `OPENTHREAD_CONFIG_MESSAGE_USE_HEAP_ENABLE` that will use `GetHeap()` to get the capacity of the internal heap for message pool. I don't have a good idea about the meaningful capacity of external heap as as there is no platform APIs for this. For now, lets keep this for `OPENTHREAD_CONFIG_MESSAGE_USE_HEAP_ENABLE`.

Reply:
It is used by `MessagePool::GetTotalBufferCount and MessagePool::GetFreeBufferCount`. I think a simple solution is to let these functions always return very large values (e.x. total=4096, free=4096) when external heap is enabled.